rust/benches/utils.rs

```diff
@@ -0,0 +1,56 @@
+use std::{
+    sync::atomic::{AtomicBool, Ordering},
+    thread,
+    time::{Duration, Instant},
+};
+
+use carmen_rust::error::BTResult;
+
+/// Executes the given operation in parallel using the specified number of threads.
+/// Thread-local data can be created using the `op_data` closure, and iteration id can be customized
+/// using the `get_id` closure.
+/// Each thread will perform its share of iterations, starting only after all threads are ready.
+/// Returns the total duration taken to complete all iterations across all threads.
+pub fn execute_with_threads<T>(
+    num_threads: u64,
+    iters: u64,
+    get_id: impl Fn(&u64) -> u64 + Send + Sync,
+    completed_iterations: &mut u64,
+    op_data: impl Fn() -> T + Send + Sync,
+    op: impl Fn(u64, &mut T) -> BTResult<(), carmen_rust::error::Error> + Send + Sync,
```

Comment:
The result is ignored, so we don't need a closure that returns a result in the first place.
```suggestion
    op: impl Fn(u64, &mut T) + Send + Sync,
```